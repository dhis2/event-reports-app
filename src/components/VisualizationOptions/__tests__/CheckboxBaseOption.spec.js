import { render } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import CheckboxBaseOption from '../Options/CheckboxBaseOption.js'

const mockStore = configureMockStore()

describe('ER > Options > CheckboxBaseOption', () => {
    it('renders checkbox with value: true, inverted: true', () => {
        const store = {
            ui: {
                options: {
                    theOptionName: true,
                },
            },
        }

        const option = {
            name: 'theOptionName',
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <CheckboxBaseOption
                    dataTest="testing-prefix"
                    inverted={true}
                    label="Should I?"
                    option={option}
                />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })

    it('renders checkbox with value: true, inverted: false', () => {
        const store = {
            ui: {
                options: {
                    theOptionName: true,
                },
            },
        }

        const option = {
            name: 'theOptionName',
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <CheckboxBaseOption
                    dataTest="testing-prefix"
                    inverted={false}
                    label="Should I?"
                    option={option}
                />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })

    it('renders checkbox with value: false, inverted: true', () => {
        const store = {
            ui: {
                options: {
                    theOptionName: false,
                },
            },
        }

        const option = {
            name: 'theOptionName',
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <CheckboxBaseOption
                    dataTest="testing-prefix"
                    inverted={true}
                    label="Should I?"
                    option={option}
                />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })

    it('renders checkbox with value: false, inverted: false', () => {
        const store = {
            ui: {
                options: {
                    theOptionName: false,
                },
            },
        }

        const option = {
            name: 'theOptionName',
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <CheckboxBaseOption
                    dataTest="testing-prefix"
                    inverted={false}
                    label="Should I?"
                    option={option}
                />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
})
