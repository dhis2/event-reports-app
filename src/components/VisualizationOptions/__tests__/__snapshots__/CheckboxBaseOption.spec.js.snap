// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ER > Options > CheckboxBaseOption renders checkbox with value: false, inverted: false 1`] = `
<div>
  <div
    class="tabSectionOption"
  >
    <label
      class="jsx-3569559994 dense"
      data-test="testing-prefix"
    >
      <input
        class="jsx-3569559994"
        name="theOptionName"
        type="checkbox"
        value=""
      />
      <div
        class="jsx-3569559994 icon dense"
      >
        <svg
          class="jsx-1744227082 jsx-2054485247 "
          viewBox="0 0 14 14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            class="jsx-1744227082 jsx-2054485247 background"
            fill="#009688"
            height="14"
            rx="2"
            width="14"
            x="0"
            y="0"
          />
          <path
            class="jsx-1744227082 jsx-2054485247 border"
            d="M2,0 L12,0 C13.1045695,0 14,0.8954305 14,2 L14,12 C14,13.1045695 13.1045695,14 12,14 L2,14 C0.8954305,14 0,13.1045695 0,12 L0,2 C0,0.8954305 0.8954305,0 2,0 Z M2,1 C1.44771525,1 1,1.44771525 1,2 L1,12 C1,12.5522847 1.44771525,13 2,13 L12,13 C12.5522847,13 13,12.5522847 13,12 L13,2 C13,1.44771525 12.5522847,1 12,1 L2,1 Z"
            fill="#004D40"
          />
          <path
            class="jsx-1744227082 jsx-2054485247 checkmark"
            d="M10.3520005,3.30015877 L5.54268293,8.03252726 L3.64634146,6.16652726 L2.28668615,7.5044281 L5.54310367,10.7010587 L11.7145993,4.628307 L10.3520005,3.30015877 Z M5.54268293,9.43547274 L5.61193184,9.36733181 L5.54226219,9.29894127 L5.47309681,9.367 L5.54268293,9.43547274 Z M10.6496329,4.41023392 L10.6495057,4.35652297 L10.5648412,4.35672356 L10.2854007,4.631693 L10.3553166,4.69984123 L10.6496329,4.41023392 Z M3.42891348,7.22439304 L3.35030905,7.22439304 L3.35034135,7.27820863 L3.64634146,7.56947274 L3.71331385,7.5035719 L3.42891348,7.22439304 Z"
            fill="#FFFFFF"
          />
          <rect
            class="jsx-1744227082 jsx-2054485247 indeterminate"
            fill="#FFFFFF"
            height="2"
            width="8"
            x="3"
            y="6"
          />
        </svg>
      </div>
      Should I?
    </label>
  </div>
</div>
`;

exports[`ER > Options > CheckboxBaseOption renders checkbox with value: false, inverted: true 1`] = `
<div>
  <div
    class="tabSectionOption"
  >
    <label
      class="jsx-3569559994 dense"
      data-test="testing-prefix"
    >
      <input
        checked=""
        class="jsx-3569559994"
        name="theOptionName"
        type="checkbox"
        value=""
      />
      <div
        class="jsx-3569559994 icon dense"
      >
        <svg
          class="jsx-1744227082 jsx-2054485247 checked"
          viewBox="0 0 14 14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            class="jsx-1744227082 jsx-2054485247 background"
            fill="#009688"
            height="14"
            rx="2"
            width="14"
            x="0"
            y="0"
          />
          <path
            class="jsx-1744227082 jsx-2054485247 border"
            d="M2,0 L12,0 C13.1045695,0 14,0.8954305 14,2 L14,12 C14,13.1045695 13.1045695,14 12,14 L2,14 C0.8954305,14 0,13.1045695 0,12 L0,2 C0,0.8954305 0.8954305,0 2,0 Z M2,1 C1.44771525,1 1,1.44771525 1,2 L1,12 C1,12.5522847 1.44771525,13 2,13 L12,13 C12.5522847,13 13,12.5522847 13,12 L13,2 C13,1.44771525 12.5522847,1 12,1 L2,1 Z"
            fill="#004D40"
          />
          <path
            class="jsx-1744227082 jsx-2054485247 checkmark"
            d="M10.3520005,3.30015877 L5.54268293,8.03252726 L3.64634146,6.16652726 L2.28668615,7.5044281 L5.54310367,10.7010587 L11.7145993,4.628307 L10.3520005,3.30015877 Z M5.54268293,9.43547274 L5.61193184,9.36733181 L5.54226219,9.29894127 L5.47309681,9.367 L5.54268293,9.43547274 Z M10.6496329,4.41023392 L10.6495057,4.35652297 L10.5648412,4.35672356 L10.2854007,4.631693 L10.3553166,4.69984123 L10.6496329,4.41023392 Z M3.42891348,7.22439304 L3.35030905,7.22439304 L3.35034135,7.27820863 L3.64634146,7.56947274 L3.71331385,7.5035719 L3.42891348,7.22439304 Z"
            fill="#FFFFFF"
          />
          <rect
            class="jsx-1744227082 jsx-2054485247 indeterminate"
            fill="#FFFFFF"
            height="2"
            width="8"
            x="3"
            y="6"
          />
        </svg>
      </div>
      Should I?
    </label>
  </div>
</div>
`;

exports[`ER > Options > CheckboxBaseOption renders checkbox with value: true, inverted: false 1`] = `
<div>
  <div
    class="tabSectionOption"
  >
    <label
      class="jsx-3569559994 dense"
      data-test="testing-prefix"
    >
      <input
        checked=""
        class="jsx-3569559994"
        name="theOptionName"
        type="checkbox"
        value=""
      />
      <div
        class="jsx-3569559994 icon dense"
      >
        <svg
          class="jsx-1744227082 jsx-2054485247 checked"
          viewBox="0 0 14 14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            class="jsx-1744227082 jsx-2054485247 background"
            fill="#009688"
            height="14"
            rx="2"
            width="14"
            x="0"
            y="0"
          />
          <path
            class="jsx-1744227082 jsx-2054485247 border"
            d="M2,0 L12,0 C13.1045695,0 14,0.8954305 14,2 L14,12 C14,13.1045695 13.1045695,14 12,14 L2,14 C0.8954305,14 0,13.1045695 0,12 L0,2 C0,0.8954305 0.8954305,0 2,0 Z M2,1 C1.44771525,1 1,1.44771525 1,2 L1,12 C1,12.5522847 1.44771525,13 2,13 L12,13 C12.5522847,13 13,12.5522847 13,12 L13,2 C13,1.44771525 12.5522847,1 12,1 L2,1 Z"
            fill="#004D40"
          />
          <path
            class="jsx-1744227082 jsx-2054485247 checkmark"
            d="M10.3520005,3.30015877 L5.54268293,8.03252726 L3.64634146,6.16652726 L2.28668615,7.5044281 L5.54310367,10.7010587 L11.7145993,4.628307 L10.3520005,3.30015877 Z M5.54268293,9.43547274 L5.61193184,9.36733181 L5.54226219,9.29894127 L5.47309681,9.367 L5.54268293,9.43547274 Z M10.6496329,4.41023392 L10.6495057,4.35652297 L10.5648412,4.35672356 L10.2854007,4.631693 L10.3553166,4.69984123 L10.6496329,4.41023392 Z M3.42891348,7.22439304 L3.35030905,7.22439304 L3.35034135,7.27820863 L3.64634146,7.56947274 L3.71331385,7.5035719 L3.42891348,7.22439304 Z"
            fill="#FFFFFF"
          />
          <rect
            class="jsx-1744227082 jsx-2054485247 indeterminate"
            fill="#FFFFFF"
            height="2"
            width="8"
            x="3"
            y="6"
          />
        </svg>
      </div>
      Should I?
    </label>
  </div>
</div>
`;

exports[`ER > Options > CheckboxBaseOption renders checkbox with value: true, inverted: true 1`] = `
<div>
  <div
    class="tabSectionOption"
  >
    <label
      class="jsx-3569559994 dense"
      data-test="testing-prefix"
    >
      <input
        class="jsx-3569559994"
        name="theOptionName"
        type="checkbox"
        value=""
      />
      <div
        class="jsx-3569559994 icon dense"
      >
        <svg
          class="jsx-1744227082 jsx-2054485247 "
          viewBox="0 0 14 14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            class="jsx-1744227082 jsx-2054485247 background"
            fill="#009688"
            height="14"
            rx="2"
            width="14"
            x="0"
            y="0"
          />
          <path
            class="jsx-1744227082 jsx-2054485247 border"
            d="M2,0 L12,0 C13.1045695,0 14,0.8954305 14,2 L14,12 C14,13.1045695 13.1045695,14 12,14 L2,14 C0.8954305,14 0,13.1045695 0,12 L0,2 C0,0.8954305 0.8954305,0 2,0 Z M2,1 C1.44771525,1 1,1.44771525 1,2 L1,12 C1,12.5522847 1.44771525,13 2,13 L12,13 C12.5522847,13 13,12.5522847 13,12 L13,2 C13,1.44771525 12.5522847,1 12,1 L2,1 Z"
            fill="#004D40"
          />
          <path
            class="jsx-1744227082 jsx-2054485247 checkmark"
            d="M10.3520005,3.30015877 L5.54268293,8.03252726 L3.64634146,6.16652726 L2.28668615,7.5044281 L5.54310367,10.7010587 L11.7145993,4.628307 L10.3520005,3.30015877 Z M5.54268293,9.43547274 L5.61193184,9.36733181 L5.54226219,9.29894127 L5.47309681,9.367 L5.54268293,9.43547274 Z M10.6496329,4.41023392 L10.6495057,4.35652297 L10.5648412,4.35672356 L10.2854007,4.631693 L10.3553166,4.69984123 L10.6496329,4.41023392 Z M3.42891348,7.22439304 L3.35030905,7.22439304 L3.35034135,7.27820863 L3.64634146,7.56947274 L3.71331385,7.5035719 L3.42891348,7.22439304 Z"
            fill="#FFFFFF"
          />
          <rect
            class="jsx-1744227082 jsx-2054485247 indeterminate"
            fill="#FFFFFF"
            height="2"
            width="8"
            x="3"
            y="6"
          />
        </svg>
      </div>
      Should I?
    </label>
  </div>
</div>
`;
